nilearn/tests/test_extmath.py

```diff
@@ -50,3 +53,31 @@ def test_is_spd_with_symmetrical_matrix():
     matrix = np.array([[2, 1],
                        [1, 1]])
     assert is_spd(matrix, verbose=0)
+
+
+def test_check_threshold():
+    matrix = np.array([[1., 2.],
+                       [2., 1.]])
+
+    name = 'threshold'
+    # few not correctly formatted strings for 'threshold'
+    wrong_thresholds = ['0.1', '10', '10.2.3%', 'asdf%']
+    for wrong_threshold in wrong_thresholds:
+        assert_raises_regex(ValueError,
+                            '{0}.+should be a number followed by '
+                            'the percent sign'.format(name),
+                            check_threshold,
+                            wrong_threshold, matrix,
+                            'fast_abs_percentile', name)
+
+    threshold = object()
+    assert_raises_regex(TypeError,
+                        '{0}.+should be either a number '
+                        'or a string'.format(name),
+                        check_threshold, threshold, matrix,
+                        'fast_abs_percentile', name)
+
+    # To check if it also gives the expected score to given threshold
+    assert_true(1. < check_threshold("50%", matrix,
+                                     percentile_calculate=fast_abs_percentile,
+                                     name=name) <= 2.)
```

Comment:
It does work with numpy scalar:
<pre>
In [15]: a = np.zeros(10)

In [17]: aa = a[0]

In [19]: aa
Out[19]: 0.0

In [20]: type(aa)
Out[20]: numpy.float64

In [24]: isinstance(aa, float)
Out[24]: True
</pre>

however, the right way to check for numbers is to do:
<pre>
import numbers
isinstance(baz, numbers.Number)
</pre>